Source/Core/AudioCommon/DPL2Decoder.cpp

```diff
@@ -96,18 +86,22 @@ static T FIRFilter(const T* buf, int pos, int len, int count, const float* coeff
 // n window length
 // w buffer for the window parameters
 */
-static void Hamming(int n, float* w)
+static std::vector<float> Hamming(int n)
 {
-  float k = float(2 * M_PI / ((float)(n - 1)));  // 2*pi/(N-1)
+  std::vector<float> w(n);
+
+  float k = static_cast<float>(2.0 * M_PI / (n - 1));
 
   // Calculate window coefficients
   for (int i = 0; i < n; i++)
-    *w++ = float(0.54 - 0.46 * cos(k * (float)i));
+    w[i] = static_cast<float>(0.54 - 0.46 * cos(k * i));
+
+  return w;
 }
 
 /******************************************************************************
-*  FIR filter design
-******************************************************************************/
+ *  FIR filter design
```

Comment:
The excessive comment header and footer can just be replaced with `// FIR filter design`